spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -540,7 +541,8 @@ private static void importUnpartitionedSparkTable(
       Configuration conf = spark.sessionState().newHadoopConf();
       MetricsConfig metricsConfig = MetricsConfig.fromProperties(targetTable.properties());
       String nameMappingString = targetTable.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
-      NameMapping nameMapping = nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+      NameMapping nameMapping = nameMappingString != null ?
+          NameMappingParser.fromJson(nameMappingString) : MappingUtil.create(targetTable.schema());
```

Comment:
I think we should change to use a name mapping by default in a separate PR. For now, let's just support a name mapping when importing ORC data.

My rationale is that I think the two are independently useful. Someone might want to change the default, but not cherry-pick ORC changes. Similarly, someone might want to use a name mapping with ORC, but not want to default to name mapping.

Also, I think that we would want to default to name mapping slightly differently. It doesn't make sense to me to create a temporary mapping that is used for metrics here, unless that mapping is also used to read the data. So I would prefer to update tables when importing and add a mapping to table metadata by default, if it is not already there.